pydicom/tests/test_dataelem.py

```diff
@@ -369,45 +356,52 @@ def test_private_repeater_tag(self):
         assert 'UN' == private_data_elem.VR
 
 
-class RawDataElementTests(unittest.TestCase):
+class TestRawDataElement(object):
+    """Tests for dataelem.RawDataElement."""
     def testKeyError(self):
         """RawDataElement: conversion of unknown tag throws KeyError..."""
         # raw data element -> tag VR length value
         #                       value_tell is_implicit_VR is_little_endian'
         # Unknown (not in DICOM dict), non-private, non-group 0 for this test
         raw = RawDataElement(Tag(0x88880002), None, 4, 0x1111,
                              0, True, True)
-        self.assertRaises(KeyError, DataElement_from_raw, raw)
+
+        with pytest.raises(KeyError, match=r"\(8888, 0002\)"):
+            DataElement_from_raw(raw)
 
     def testValidTag(self):
         """RawDataElement: conversion of known tag succeeds..."""
         raw = RawDataElement(Tag(0x00080020), 'DA', 8, b'20170101',
                              0, False, True)
         element = DataElement_from_raw(raw, default_encoding)
-        self.assertEqual(element.name, 'Study Date')
-        self.assertEqual(element.VR, 'DA')
-        self.assertEqual(element.value, '20170101')
+        assert element.name == 'Study Date'
+        assert element.VR == 'DA'
+        assert element.value == '20170101'
 
         raw = RawDataElement(Tag(0x00080000), None, 4, b'\x02\x00\x00\x00',
                              0, True, True)
         elem = DataElement_from_raw(raw, default_encoding)
         assert elem.VR == 'UL'
 
-    @unittest.skipIf(sys.version_info >= (3, ), 'Testing Python 2 behavior')
+    @pytest.mark.skipif(sys.version_info >= (3, ), reason='Python 2 behavior')
     def testTagWithoutEncodingPython2(self):
         """RawDataElement: no encoding needed in Python 2."""
         raw = RawDataElement(Tag(0x00104000), 'LT', 23,
                              b'comment\\comment2\\comment3',
                              0, False, True)
         element = DataElement_from_raw(raw)
-        self.assertEqual(element.name, 'Patient Comments')
+        assert element.name == 'Patient Comments'
 
-    @unittest.skipIf(sys.version_info < (3, ), 'Testing Python 3 behavior')
+    @pytest.mark.skip(reason="Not testing what is claimed")
+    @pytest.mark.skipif(sys.version_info < (3, ), reason='Python 3 behavior')
     def testTagWithoutEncodingPython3(self):
         """RawDataElement: raises if no encoding given in Python 3."""
```

Comment:
@mrbean-bremen do you know what this test is supposed to be checking? It was raising a `TypeError` because of `RawDataElement()()` (ie it was attempting to call RawDataElement()) not because of an encoding failure.

Reply:
Nevermind, got it. Python2.7...

Reply:
I have not the slightest idea. It should have tested the presence of the `encoding` parameter in `DataElement_from_raw`. First, the test has been wrong from the beginning (as you saw), and second, I think it is not valid anymore after I have changed the encoding parameter (whch used to be a triple of values).
I would delete it, remove the condition for `testTagWithoutEncodingPython2` and rename it to something like `test_data_element_without_encoding`.

Reply:
Thanks, will do